relayer/application_relayer.go

```diff
@@ -317,6 +373,38 @@ func (r *ApplicationRelayer) RelayerID() database.RelayerID {
 	return r.relayerID
 }
 
+// getPChainHeightForValidatorSet determines the appropriate P-Chain height for validator set selection
+// Returns ProposedHeight for current validators if Granite is not activated, or the epoch P-Chain height if activated
+func (r *ApplicationRelayer) getPChainHeightForValidatorSet(ctx context.Context) (uint64, error) {
```

Comment:
This name confused me a little at first. I would probably prefer something like `getPChainHeightForDestination`